File: csharp/AoC2021/Day25/Day25/SeaCucumber.cs
namespace Day25;

public enum SeaCucumber
{
    East,
    South,
    Empty,
}

public static class SeaCucumberExtensions
{
    public static bool IsEast(this SeaCucumber sc) => sc == SeaCucumber.East;
    public static bool IsEmpty(this SeaCucumber sc) => sc == SeaCucumber.Empty;
}

File: rust/2021/day25/src/sea_cucumber.rs
pub enum SeaCucumber {
    East,
    South,
    Empty,
}

impl SeaCucumber {
    pub fn is_east(&self) -> bool {
        matches!(self, SeaCucumber::East)
    }

    pub fn is_empty(&self) -> bool {
        matches!(self, SeaCucumber::Empty)
    }
}

impl Default for SeaCucumber {
    fn default() -> Self {
        SeaCucumber::Empty
    }
}
